recipes/vtk-m/meta.yaml

```diff
@@ -0,0 +1,63 @@
+{% set name = "vtk-m" %}
+{% set version_major = "2" %}
+{% set version_minor = "0" %}
+{% set version_patch = "0" %}
+{% set version = version_major + "." + version_minor + "." + version_patch %}
+{% set common_flags = "-DCMAKE_BUILD_TYPE=Release -DVTKm_ENABLE_EXAMPLES=ON -DVTKm_INSTALL_EXAMPLES=OFF" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://gitlab.kitware.com/vtk/vtk-m/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz
+  sha256: 32643cf3564fa77f8e2a2a5456a574b6b2355bb68918eb62ccde493993ade1a3
+
+build:
+  script:
+    - cmake -GNinja -S $SRC_DIR  -B build -DCMAKE_INSTALL_PREFIX=$PREFIX {{ common_flags }}  # [unix]
+    - cmake -GNinja -S %SRC_DIR% -B build -DCMAKE_INSTALL_PREFIX=%PREFIX% {{ common_flags }}  # [win]
+    - cmake --build build -j
+    - ./build/examples/smoke_test/smoke_test  # [unix]
+    - build\examples\smoke_test\smoke_test.exe  # [win]
+    - cmake --install build
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - ninja
+    - pkg-config
+    - gnuconfig
+    - cmake =3.23
+
+test:
```

Comment:
Please test for the existence of all the install shared libraries, and one of the headers. You can use a jinja loop to reduce repeated commands.

Reply:
This is tricky, VTK-m produce several shlibs depending on the flags/version of the project. Is this a requirement for C/C++ packages?  Would it satisfy to check the core shlibs? in Spack there was certain push for this for it was eventually forgo due to the difficulty in large project.

Reply:
To concretize further here is the list of a minimal shlibs generated by VTK-m:

```
→ ls lib/
cmake/                                            libvtkm_filter_contour-2.0.so.2.0                 libvtkm_filter_geometry_refinement-2.0.so         libvtkm_filter_vector_analysis-2.0.so.2.0.0
libvtkm_cont-2.0.so                               libvtkm_filter_contour-2.0.so.2.0.0               libvtkm_filter_geometry_refinement-2.0.so.2.0     libvtkm_filter_zfp-2.0.so
libvtkm_cont-2.0.so.2.0                           libvtkm_filter_core-2.0.so                        libvtkm_filter_geometry_refinement-2.0.so.2.0.0   libvtkm_filter_zfp-2.0.so.2.0
libvtkm_cont-2.0.so.2.0.0                         libvtkm_filter_core-2.0.so.2.0                    libvtkm_filter_image_processing-2.0.so            libvtkm_filter_zfp-2.0.so.2.0.0
libvtkm_cont_testing-2.0.so                       libvtkm_filter_core-2.0.so.2.0.0                  libvtkm_filter_image_processing-2.0.so.2.0        libvtkm_io-2.0.so
libvtkm_cont_testing-2.0.so.2.0                   libvtkm_filter_density_estimate-2.0.so            libvtkm_filter_image_processing-2.0.so.2.0.0      libvtkm_io-2.0.so.2.0
libvtkm_cont_testing-2.0.so.2.0.0                 libvtkm_filter_density_estimate-2.0.so.2.0        libvtkm_filter_mesh_info-2.0.so                   libvtkm_io-2.0.so.2.0.0
libvtkmdiympi_nompi.so                            libvtkm_filter_density_estimate-2.0.so.2.0.0      libvtkm_filter_mesh_info-2.0.so.2.0               libvtkm_rendering-2.0.so
libvtkmdiympi_nompi.so.2.0                        libvtkm_filter_entity_extraction-2.0.so           libvtkm_filter_mesh_info-2.0.so.2.0.0             libvtkm_rendering-2.0.so.2.0
libvtkmdiympi_nompi.so.2.0.0                      libvtkm_filter_entity_extraction-2.0.so.2.0       libvtkm_filter_multi_block-2.0.so                 libvtkm_rendering-2.0.so.2.0.0
libvtkmdiympi.so                                  libvtkm_filter_entity_extraction-2.0.so.2.0.0     libvtkm_filter_multi_block-2.0.so.2.0             libvtkm_rendering_testing-2.0.so
libvtkmdiympi.so.2.0                              libvtkm_filter_field_conversion-2.0.so            libvtkm_filter_multi_block-2.0.so.2.0.0           libvtkm_rendering_testing-2.0.so.2.0
libvtkmdiympi.so.2.0.0                            libvtkm_filter_field_conversion-2.0.so.2.0        libvtkm_filter_resampling-2.0.so                  libvtkm_rendering_testing-2.0.so.2.0.0
libvtkm_filter_clean_grid-2.0.so                  libvtkm_filter_field_conversion-2.0.so.2.0.0      libvtkm_filter_resampling-2.0.so.2.0              libvtkm_source-2.0.so
libvtkm_filter_clean_grid-2.0.so.2.0              libvtkm_filter_field_transform-2.0.so             libvtkm_filter_resampling-2.0.so.2.0.0            libvtkm_source-2.0.so.2.0
libvtkm_filter_clean_grid-2.0.so.2.0.0            libvtkm_filter_field_transform-2.0.so.2.0         libvtkm_filter_scalar_topology-2.0.so             libvtkm_source-2.0.so.2.0.0
libvtkm_filter_connected_components-2.0.so        libvtkm_filter_field_transform-2.0.so.2.0.0       libvtkm_filter_scalar_topology-2.0.so.2.0         libvtkm_worklet-2.0.so
libvtkm_filter_connected_components-2.0.so.2.0    libvtkm_filter_flow-2.0.so                        libvtkm_filter_scalar_topology-2.0.so.2.0.0       libvtkm_worklet-2.0.so.2.0
libvtkm_filter_connected_components-2.0.so.2.0.0  libvtkm_filter_flow-2.0.so.2.0                    libvtkm_filter_vector_analysis-2.0.so             libvtkm_worklet-2.0.so.2.0.0
libvtkm_filter_contour-2.0.so                     libvtkm_filter_flow-2.0.so.2.0.0                  libvtkm_filter_vector_analysis-2.0.so.2.0         
```

Reply:
Two thirds of these listed items are links. For example, you can only test for `libvtkmdiympi.so.2.0.0` and not `libvtkmdiympi.so.2.0` or `libvtkmdiympi.so`.

```
{% for each_mod in vtk_mods %}
- test -f $PREFIX/lib/libvtkm{{ each_mod }}.so.{{ version }}     # [linux]
- test -f $PREFIX/lib/libvtkm{{ each_mod }}.{{ version }}.dylib  # [osx]
{% endfor %}
```

> VTK-m produce several shlibs depending on the flags/version of the project.

This seems like a reason *for* testing the outputs if they change depending on various build flags. You'll want to confirm that you set the correct flags.